conda/base/context.py

```diff
@@ -520,6 +518,20 @@ def post_build_validation(self):
                 "Only one can be set to 'True'.",
             )
             errors.append(error)
+        if not self.channels:
+            from conda.core.prefix_data import PrefixData
+            from conda.models.channel import Channel
+
+            conda_meta_info = next(PrefixData(self.root_prefix).query("conda"), None)
+            if conda_meta_info:
+                conda_channel = Channel(conda_meta_info.channel).canonical_name
+            else:
+                conda_channel = "[desired channel]"
+            log.warning(
+                "Channel list is empty. Please run 'conda config --add channels %s' to "
+                "remove this warning.",
+                conda_channel,
```

Comment:
Just note that this'll require more info I think, but the word-smithing isn't important at this stage